components/datetime/src/provider/calendar/mod.rs

```diff
@@ -15,19 +15,23 @@ use icu_provider::{yoke, zerofrom};
 pub use skeletons::*;
 pub use symbols::*;
 
-#[icu_provider::data_struct(marker(
-    DateLengthsV1Marker,
-    "datetime/datelengths@1",
-    extension_key = "ca"
-))]
+#[icu_provider::data_struct(
+    marker(GregorianDateLengthsV1Marker, "datetime/gregory/datelengths@1"),
+    marker(BuddhistDateLengthsV1Marker, "datetime/buddhist/datelengths@1"),
+    marker(JapaneseDateLengthsV1Marker, "datetime/japanese/datelengths@1"),
+    marker(JapaneseExtendedDateLengthsV1Marker, "datetime/japanext/datelengths@1"),
+    marker(CopticDateLengthsV1Marker, "datetime/coptic/datelengths@1"),
+    marker(IndianDateLengthsV1Marker, "datetime/indian/datelengths@1"),
+    marker(EthiopicDateLengthsV1Marker, "datetime/ethiopic/datelengths@1")
```

Comment:
suggestion/thought: `datelengths/calendar` or `calendar/datelengths`? I feel like the former is nicer but I'm not sure.

The former gives a clear expectation that all calendars have the same formatting data key structure.

Reply:
Could go either way. With calendar first, it means you get one directory per calendar, and the "foo@1" is the same as the data struct name FooV1, which is nice.